homeassistant/components/conversation/session.py

```diff
@@ -326,3 +362,91 @@ async def async_update_llm_data(
             agent_id=user_input.agent_id,
             content=prompt,
         )
+
+    async def async_get_tools(self) -> list[llm.Tool] | None:
+        """Get the tools for the agent."""
+        if not self.llm_api:
+            return None
+        return self.llm_api.tools
+
+    async def _async_call_tools(
+        self,
+        agent_id: str | None,
```

Comment:
We shouldn't have to pass in `agent_id`. I think that we should have a ChatSession with an active conversation with the ConversationInput that started this conversation. That input contains all the useful info.